eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionReportWriter.java

```diff
@@ -0,0 +1,130 @@
+        ISubmittedVariant variant = denormalizeVariant(normalizedVariant);
+        String vcfLine = variantToVcfLine(id, variant);
+        fileWriter.write(vcfLine);
+        fileWriter.newLine();
+    }
+
+    private ISubmittedVariant denormalizeVariant(ISubmittedVariant normalizedVariant) {
+        if (normalizedVariant.getReferenceAllele().isEmpty() || normalizedVariant.getAlternateAllele().isEmpty()) {
+            if (fastaSequenceReader.doesContigExist(normalizedVariant.getContig())) {
+                return createVariantWithContextBase(normalizedVariant);
+            } else {
+                throw new IllegalArgumentException("Contig '" + normalizedVariant.getContig()
+                                                           + "' does not appear in the fasta file ");
+            }
+        } else {
+            return normalizedVariant;
+        }
+    }
+
+    private ISubmittedVariant createVariantWithContextBase(ISubmittedVariant normalizedVariant) {
+        long newStart;
+        String newReference;
+        String newAlternate;
+        if (normalizedVariant.getStart() == 1) {
+            // VCF 4.2 section 1.4.1.4. REF: "the REF and ALT Strings must include the base before the event unless the
+            // event occurs at position 1 on the contig in which case it must include the base after the event"
+            newStart = normalizedVariant.getStart() + 1;
+            String contextBase = fastaSequenceReader.getSequence(normalizedVariant.getContig(), newStart, newStart);
+            newReference = normalizedVariant.getReferenceAllele() + contextBase;
+            newAlternate = normalizedVariant.getAlternateAllele() + contextBase;
+        } else {
+            newStart = normalizedVariant.getStart() - 1;
+            String contextBase = fastaSequenceReader.getSequence(normalizedVariant.getContig(), newStart, newStart);
+            newReference = contextBase + normalizedVariant.getReferenceAllele();
+            newAlternate = contextBase + normalizedVariant.getAlternateAllele();
+        }
+
+        return new SubmittedVariant(normalizedVariant.getAssemblyAccession(),
+                                    normalizedVariant.getTaxonomyAccession(),
+                                    normalizedVariant.getProjectAccession(),
+                                    normalizedVariant.getContig(),
+                                    newStart,
+                                    newReference,
+                                    newAlternate,
+                                    normalizedVariant.isSupportedByEvidence());
+    }
+
+    protected String variantToVcfLine(Long id, ISubmittedVariant variant) {
+        String variantLine = String.join("\t",
+                                         variant.getContig(),
+                                         Long.toString(variant.getStart()),
+                                         accessionPrefix + Long.toString(id),
```

Comment:
Since you are using the "+" sign and accessionPrefix is an String you don't need to call the toString() method for id